examples/fungible/src/service.rs

```diff
@@ -31,13 +31,26 @@ impl Service for FungibleToken {
         _context: &QueryContext,
         request: Request,
     ) -> Result<Response, Self::Error> {
+        let query_root = FungibleExtendedView(self.clone(), FungibleViewExtension);
         let schema =
-            Schema::build(self.clone(), Operation::mutation_root(), EmptySubscription).finish();
+            Schema::build(query_root, Operation::mutation_root(), EmptySubscription).finish();
         let response = schema.execute(request).await;
         Ok(response)
     }
 }
 
+struct FungibleViewExtension;
```

Comment:
Rather than going through the hassle of `MergedObject`, you can use `ComplexObject` to add calculated fields to a `SimpleObject`: https://docs.rs/async-graphql/latest/async_graphql/attr.ComplexObject.html

Reply:
The problem is that if you specify `#[graphql(complex)]` you _have_ to provide a `ComplexObject` implementation, so we'd have to do that in both the service _and_ the contract (we can't do it in `state.rs` directly, because `parameters` is a method of `Service` and `Contract`, and in `state.rs` we can't use either). Not sure if it's really cleaner if we have to give an empty `ComplexObject` implementation in `contract.rs`.

Reply:
Or: Import both traits in `state.rs` and ignore the warning that one of them is unused…?

Reply:
I think it's still a bit cleaner to do the empty `ComplexObject` since even though it adds a little boilerplate it leaves the rest of the code clean, whereas `MergedObject` requires understanding the structure of the merged object (and how `async-graphql::MergedObject` works) to figure out what's going on.  I think generally shallower nesting is better.

Reply:
Done; you're right, this is probably cleaner.

Reply:
Bit of a nit now, but given that this is a user-facing example I might like to see a comment explaining why there's an empty `ComplexObject` impl in case people are confused.  Also reading it should elucidate a little bit how the contract/service split works in the example.